src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1217,6 +1219,34 @@ public override int GetHashCode()
             return (this.Name.GetHashCode());
         }
 
+        internal static unsafe int GetIgnoreCaseHash(string source)
+        {
+            Debug.Assert(source != null, "source must not be null");
+
+            // Do not allocate on the stack if string is empty
+            if (source.Length == 0)
+            {
+                return source.GetHashCode();
+            }
+
+            char[] borrowedArr = null;
+            Span<char> span = source.Length <= 255 ?
+                stackalloc char[255] :
+                (borrowedArr = ArrayPool<char>.Shared.Rent(source.Length));
+
+            source.AsSpan().ToUpperInvariant(span);
+
+            // Slice the array to the input size as we could have gotten a larger array from the ArrayPool.
```

Comment:
It would be more appropriate to use the length returned by ToUpperInvariant for the slice, not the source Length.